### PR TITLE
Make timefactory an anonymous class

### PR DESCRIPTION
**Pull request by ChristophWurst**
I wanted to play with anonymous classes in Nextcloud and figured this would be a perfect candidate to give a try.

I've first cleaned up the usage of the time factory implementation and replaced it with the interface. Secondly, I've removed the implementation as separate class and move it to the service registration.

Let me know what you think. In theory this saves a bit of disk access as one file has to be loaded less, but then it also kind of moves code (here, it's just a one-liner) into the service registration.

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Codewise this makes a lot of sense 👍

File: lib/private/AppFramework/DependencyInjection/DIContainer.php [hunk 1]
```diff
@@ -127,9 +127,6 @@ public function __construct($appName, $urlParams = array(), ServerContainer $ser
 		});
 		$this->registerAlias('Request', IRequest::class);
 
-		$this->registerAlias(\OCP\AppFramework\Utility\ITimeFactory::class, \OC\AppFramework\Utility\TimeFactory::class);
```
Comment (ChristophWurst) [inline lib/private/AppFramework/DependencyInjection/DIContainer.php]: Interestingly, these lines were useless :man_shrugging: 

I wonder if we could remove the others as well which are already defined in the server container and what consequences that would have.

File: lib/private/Server.php [hunk 2]
```diff
@@ -1123,7 +1122,13 @@ public function __construct($webRoot, \OC\Config $config) {
 		$this->registerAlias(\OCP\AppFramework\Utility\IControllerMethodReflector::class, \OC\AppFramework\Utility\ControllerMethodReflector::class);
 		$this->registerAlias('ControllerMethodReflector', \OCP\AppFramework\Utility\IControllerMethodReflector::class);
 
-		$this->registerAlias(\OCP\AppFramework\Utility\ITimeFactory::class, \OC\AppFramework\Utility\TimeFactory::class);
+		$this->registerService(\OCP\AppFramework\Utility\ITimeFactory::class, function (Server $c) {
```
Comment (rullzer) [inline lib/private/Server.php]: Maybe move it to a function in Server 'anonymous classes' or something? Just so we know where to look.

**Comment by rullzer** [comment]
I like it. Especially for such small classes this feels good.

File: lib/private/Server.php (hunk 2 above)
Comment (ChristophWurst) [reply lib/private/Server.php]: What do you mean by that? A function that is called here that registers those anonymous services?

Comment (rullzer) [reply lib/private/Server.php]: yeah. Or pff maybe this is fine as well. I just kinda dislike that huge function to register it all ;)

**Comment by icewind1991** [comment]
I'm not a big fan of this, I don't feel it really ads anything and just makes it harder to understand whats going on

**Comment by ChristophWurst** [comment]
Soooo :+1: or :-1:

**Comment by icewind1991** [comment]
:-1: from me in this case

**Pull request closed without merging**